ci/run_python_tests.sh

```diff
@@ -0,0 +1,171 @@
+commit_sha="$(jq -rc '.check_runs[0].head_sha' <<< "${check_runs}")"
+MAINTAINER_KEYS="219f55740bbf7a1ce368ba45fb7053ce4991b669 8EDE560493C65AC1 D3A22623FF9B9F11 1FCA7260796CB902 F136264D7F4A2BB5"
+
+travis_urls="
+#amd64
+https://f001.backblazeb2.com/file/ci-horizen/amd64-linux-ubuntu_focal-${travis_build_id}-${commit_sha}.tar.gz.sha256
+https://f001.backblazeb2.com/file/ci-horizen/amd64-linux-ubuntu_focal-${travis_build_id}-${commit_sha}.tar.gz
+"
+
+
+# Step 2
+echo "" && echo "=== Create folder structure ===" && echo ""
+
+base_dir="${CURRENT_DIR}/zen_release_${zen_tag}"
+CACHE_DIR="$base_dir"
+is_cached=false
+
+if [ -d "${base_dir}" ]; then
+  echo "${base_dir} folder already exists, using cache!"
+  cd "${base_dir}"/travis_files
+  is_cached=true
+
+else
+  mkdir -p "${base_dir}"/{travis_files,src}
+
+  # Step 3
+  echo "" && echo "=== Download artifacts ===" && echo ""
+
+  cd "${base_dir}"/travis_files
+  echo "$travis_urls" > ./travis_urls.txt
+  sudo apt-get update
+  sudo apt-get -y --no-install-recommends install aria2
+  aria2c -x16 -s16 -i ./travis_urls.txt --allow-overwrite=true --always-resume=true --auto-file-renaming=false
+
+fi
+
+
+# Step 4
+echo "" && echo "=== Checksum verification===" && echo ""
+
+if shasum -a256 -c ./*.sha256; then
+  echo "Checksum verification passed."
+else
+  fn_die "Checksum verification failed."
+fi
+
+# Step 5
+if [ "$is_cached" = false ]; then
+  echo "" && echo "=== Extract artifacts from tar" && echo ""
+  tar_file="$(find "$(realpath ${base_dir}/travis_files/)" -type f -name "*.tar.gz")"
+
+  release_folder="zen-${zen_tag}-amd64"
+  mkdir -p "${base_dir}/src/${release_folder}"
+  tar -xzf "${tar_file}" -C "${base_dir}/src/${release_folder}"
+fi
+
+export CACHE_DIR
+
+# Step 6
+echo "" && echo "=== Verify git tag signed by allowlisted maintainer" && echo ""
+
+cd "${base_dir}/src/${release_folder}"
+GNUPGHOME="$(mktemp -d 2>/dev/null || mktemp -d -t "GNUPGHOME")"
+export GNUPGHOME
+import_gpg_keys "${MAINTAINER_KEYS}"
+check_signed_tag "${zen_tag}" && IS_RELEASE="true" || IS_RELEASE="false"
+export IS_RELEASE
+( gpgconf --kill dirmngr || true )
+( gpgconf --kill gpg-agent || true )
+rm -rf "${GNUPGHOME:?}"
+unset GNUPGHOME
+
+# Step 7
+echo "" && echo "=== Export BITCOINCLI, BITCOIND and SIDECHAIN_SDK path as env vars, needed for python tests" && echo ""
+
+BITCOINCLI="${base_dir}/src/${release_folder}/src/zen-cli"
+BITCOIND="${base_dir}/src/${release_folder}/src/zend"
+SIDECHAIN_SDK="${CURRENT_DIR}"
+
+if [[ ! -f "$BITCOINCLI" ]]; then
+  fn_die "zen-cli does not exist in the given path. Exiting ..."
+fi
+if [[ ! -f "$BITCOIND" ]]; then
+  fn_die "zend does not exist in the given path. Exiting ..."
+fi
+if [[ ! -d "$SIDECHAIN_SDK" ]]; then
+  fn_die "Sidechain-SDK does not exist in the given path. Exiting ..."
+fi
+
+export BITCOINCLI
+export BITCOIND
+export SIDECHAIN_SDK
+
+# Step 8
+echo "" && echo "=== Fetch zen params" && echo ""
```

Comment:
Non-blocking, the params could be added as an additional travis cache folder. fetch-params.sh is safe to run if the files are already downloaded, it will just skip downloading in that case.

Reply:
Actually this will happen already because the whole folder is cached.